lamindb/_record.py

```diff
@@ -168,68 +169,98 @@ def suggest_records_with_similar_names(
 
 
 def __init__(record: Record, *args, **kwargs):
-    if not args:
-        validate_fields(record, kwargs)
-
-        # do not search for names if an id is passed; this is important
-        # e.g. when synching ids from the notebook store to lamindb
-        has_consciously_provided_uid = False
-        if "_has_consciously_provided_uid" in kwargs:
-            has_consciously_provided_uid = kwargs.pop("_has_consciously_provided_uid")
-        if (
-            isinstance(
-                record, (CanCurate, Collection, Transform)
-            )  # Collection is only temporary because it'll get a key field
-            and settings.creation.search_names
-            and not has_consciously_provided_uid
-        ):
-            name_field = getattr(record, "_name_field", "name")
-            exact_match = suggest_records_with_similar_names(record, name_field, kwargs)
-            if exact_match is not None:
-                if "version" in kwargs:
-                    if kwargs["version"] is not None:
-                        version_comment = " and version"
-                        existing_record = record.__class__.filter(
-                            **{
-                                name_field: kwargs[name_field],
-                                "version": kwargs["version"],
-                            }
-                        ).one_or_none()
+    try:
```

Comment:
Because the code below is already heavily indented and deserves to be refactored, can you please separate this whole new try except logic into a new function?